### ISSUE TITLE
Using groovy changeset causes an UnknownChangelogFormatException

### ISSUE THREAD

**Issue by ericdalling** [jenkinsci/liquibase-runner-plugin#11]
Is it possible to run a groovy changeset with this plugin?  When I try, get the following exception:
liquibase.exception.UnknownChangelogFormatException: Cannot find parser that supports changeset.groovy

**Comment by prospero238** [comment]
Eric, 

The short answer is "no".

The slightly longer answer is "probably not, but maybe".  Could you educate me a bit on your project and liquibase set up?  I'm having trouble using the groovy extension locally.  My steps are below:

- I downloaded the extension jar from maven central ([here](http://www.mvnrepository.com/artifact/org.liquibase/liquibase-groovy-dsl/1.2.1)) and placed it in LIQUIBASE_HOME/lib.
- I created a very simple changelog file named "changelog.groovy"
- I execute liquibase with the above changelog file.

No matter what version of liquibase I use, liquibase seems to always yield the error "Cannot find parser that supports changelog.groovy".

Theoretically, you should be able to configure your Jenkins job such that the Liquibase builder's classpath field contains the path to any liquibase extension jar.  But since I can't seem to get the groovy extension to work locally, there's no hope that the plugin would work either.

Also please note that the liquibase runner plugin uses Liquibase 3.5.1, whereas I see the groovy extension only appears to support liquibase 3.4.2 (maybe its still compatible?).

I can potentially add a feature which allows groovy DSLs, but I'd need a working example first!

Thank you for your time & attention.

**Comment by ericdalling** [comment]
Thanks for your reply. In the past we've used Liquibase with Grails projects and so we are used to writing the changelogs in Groovy. We would then use either Grails or Gradle to apply them. The current project that I'm working on isn't using Grails/Gradle but is a Java/Maven application. As I always prefer writing things in Groovy over XML, I tried going that route first but couldn't get them to work. I also tried using Yaml (still preferable to XML) with this plugin but I got the same parsing exception as with Groovy. The only way I could get it to work was to use standard XML changelogs. Since I've always used either Grails or Gradle with the Groovy changelogs, I've never tried to wire up Liquibase directly with the Groovy scripts. I think for what we are doing with this project, the XML scripts will be fine but I thought I would check to see what else was supported.

**Comment by prospero238** [comment]
Eric,

Many thanks for the information.  A couple of notes:

*  Based on my not-very-thorough research, my sense is that groovy-formatted changesets only really worked when using with gradle (and never worked when using "naked" liquibase).  It sounds as if this matches your experience (i.e.  your groovy changesets were handled by gradle (and/or perhaps grails)).

* As far as I'm aware, YAML support is supported by the liquibase runner plugin  (there's even a [specific test](https://github.com/jenkinsci/liquibase-runner-plugin/blob/develop/src/test/java/org/jenkinsci/plugins/liquibase/integration/ChangesetEvaluatorBuildResultTest.java#L86) for handling yaml files.  The yaml file I use is [here](https://github.com/jenkinsci/liquibase-runner-plugin/blob/develop/src/test/resources/example-changesets/yaml-changeset-sunnyday.yml).).  If you're seeing an error using yaml, I'd definitely like to know more.  The "unknown" format exception, I believe, is usually thrown based on the changelog's file extension.  Does your change log file end with ".yml" or ".yaml"?

To summarize, short of (re)-creating a groovy extension for liquibase, I don't see a way to handle groovy changesets.  YAML should be handled right now, and if you've uncovered a bug with it, I'd love to hear more.  

I share your disinclination towards XML.  There's also JSON, but really YAML should work (and in my opinion the better option).

**Comment by robertericreeves** [comment]
We changed the Liquibase Runner Plugin for 1.3.0 significantly. That should resolve this issue.
